domain/tutoringModel/nestor.py

```diff
@@ -0,0 +1,281 @@
+# LIBARIES:
+# import argparse
+# import sys
+# import time
+import os
+import random
+
+from pgmpy.inference import VariableElimination
+from pgmpy.readwrite import XMLBIFReader
+
+from domain.tutoringModel.utils import ran_seed
+from errors import errors as err
+
+
+class Nestor:
+    """
+    This class is defined to implement the
+    CausalNestor algorithm
+    """
+
+    def __init__(self):
+        """
+        All the global values used in Nestor
+        are initiated here.
+        """
+        random.seed(ran_seed)
+        # the following are the LE formats used in RGB
+        self.rgb_le_variables = [
+            "CT",
+            "BO",
+            "LG",
+            "MS",
+            "QU",
+            "EX",
+            "SU",
+            "AAM",
+            "VAM",
+            "TAM",
+        ]
+        # Loading the saved BN from local disk
+        # self.path_to_model = os.path.join(
+        #     "Nestor",
+        #     "name3_saved_bn.xml"
+        # )
+        # In case BFI and/or LIST-k is used
+        # this dict is used to decode the numerical values
+        self.likert_scale = {
+            1: "strong_disagree",
+            2: "disagree",
+            3: "neither_agree_disagree",
+            4: "agree",
+            5: "strong_agree",
+        }
+        # the following dict is used to map
+        # output variables' states of trained BN with LE formats
+        # used in HASKI.
+        self.le_name_map_rgb_to_common_HASKI = {
+            "{'CT': 'Yes'}": ["FO"],
+            "{'CT': 'No'}": ["FO"],
+            "{'BO': 'Yes'}": ["KÜ"],
```

Comment:
I defined all the constants in utils.constans.py and using them in tutoringModel.nestor.py, except for "le_name_map_rgb_to_common_HASKI" due to the following reasons

1. This particular dict is not a constant, since it uses a random module to pick up learning elements, in case of multiple mappings. 
2. I even tried defining this dict in utils.constants.py but it is leading to "ImportError: most likely due to circular import". When this particular dict is defined in utils.constants.py. Both the utils.constants.py and domain.tutoringModel.nestor.py itself use the random_seed_gen set by the Unix timestamp (so, utils.constants.py imports random_gen and tutoringModel.nestor.py itself imports random_gen causing error). 

So, I would prefer defining this dict (which is not constant and varies) globally in the Nestors' __init__ method. 
Do you have other ideas :)?

Reply:
If it's ok, we can resolve this conversation!

Reply:
You could use it like this to at least have some forms of constants:
"{'BO': 'Yes'}": [cons.abbreviation_ct],
while having import utils.constants as cons

This also propagates to the test files (you can replace "KÜ" etc. with cons.abbreviation_ct). I tried running the tests and they still work after this change^^